src/main/java/org/candlepin/controller/Refresher.java

```diff
@@ -173,9 +184,18 @@ public void run() {
         }
 
         for (Owner owner : this.owners.values()) {
-            poolManager.refreshPoolsWithRegeneration(this.subAdapter, this.prodAdapter, owner, this.lazy);
-            recalculatePoolQuantitiesForOwner(owner);
-            updateRefreshDate(owner);
+            try {
+                poolManager.refreshPoolsWithRegeneration(this.subAdapter, this.prodAdapter, owner, this.lazy);
+                recalculatePoolQuantitiesForOwner(owner);
+                updateRefreshDate(owner);
+            }
+            catch (SubscriptionServiceException e) {
+                throw new CandlepinException(null,
```

Comment:
We shouldn't be using CandlepinExceptions at this level, as we're not API-facing. These should either be rethrown directly or wrapped in something like a RuntimeException or something more specific if applicable.